src/ucp/core/ucp_ep.c

```diff
@@ -781,28 +781,29 @@ void ucp_ep_cleanup_lanes(ucp_ep_h ep)
 
     for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
         uct_ep = ep->uct_eps[lane];
-        if (uct_ep != NULL) {
-            ucs_debug("ep %p: purge uct_ep[%d]=%p", ep, lane, uct_ep);
-            uct_ep_pending_purge(uct_ep, ucp_destroyed_ep_pending_purge, ep);
-        }
-    }
-
-    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
-        uct_ep = ep->uct_eps[lane];
-        if (uct_ep == NULL) {
-            continue;
-        }
-
-        ucs_debug("ep %p: destroy uct_ep[%d]=%p", ep, lane, uct_ep);
-        uct_ep_destroy(uct_ep);
-    }
 
-    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
-        /* set UCT EP to failed UCT EP to make sure if UCP EP won't be destoyed
+        /* set UCT EP to failed UCT EP to make sure if UCP EP won't be destroyed
          * due to some UCT EP discarding procedures are in-progress and UCP EP
          * may get some operation completions which could try to dereference
          * its lanes */
         ep->uct_eps[lane] = &ucp_failed_tl_ep;
+
+        if ((uct_ep == NULL) || (uct_ep == &ucp_failed_tl_ep)) {
```

Comment:
why need to check for ucp_failed_tl_ep?
its purge/destroy should be NOP anyway

Reply:
if not check and add the following assertion:
```
ucs_assert((uct_ep == &ucp_failed_tl_ep) || (ep->ref_cnt == 1));
```
then Coverity fails and thinks that `uct_ep_destroy()` for `ucp_failed_tl_ep` can be `ucp_proxy_ep_destroy()`
I think better to avoid suppressing such warnings

Reply:
can we assert it's not ucp_proxy_ep_destroy to silence coverity?
prefer that coveriy fixes will have less effect on flow

Reply:
ok, will do

Reply:
done